iopath/common/azure_blob.py

```diff
@@ -0,0 +1,631 @@
+            container=container, blob=blob
+        ).get_blob_properties()
+
+        return {k: v for k, v in props.items()}
+
+    def _enumerate_blobs(self, path: str) -> Generator[str, None, None]:
+        account, container, path_prefix = self._parse_uri(path)
+        client = self._get_client(account)
+        return client.get_container_client(container).list_blobs(
+            name_starts_with=path_prefix
+        )
+
+    def _exists(self, path: str, **kwargs: Any) -> bool:
+        """
+        Checks if there is a resource at the given URI.
+        Args:
+            path (str): A URI supported by this PathHandler
+        Returns:
+            bool: true if the path exists
+        """
+        self._check_kwargs(kwargs)
+        return self._isfile(path) or self._isdir(path)
+
+    def _isfile(self, path: str, **kwargs: Any) -> bool:
+        """
+        Checks if the resource at the given URI is a file.
+        Args:
+            path (str): A URI supported by this PathHandler
+        Returns:
+            bool: true if the path is a file
+        """
+        self._check_kwargs(kwargs)
+
+        try:
+            props = self._get_blob_properties(path)
+            return props is not None
+        except azure_exceptions.AzureError as e:
+            logger.exception(e)
+            return False
+
+    def _isdir(self, path: str, **kwargs: Any) -> bool:
+        """
+        Checks if the resource at the given URI is a directory.
+        Args:
+            path (str): A URI supported by this PathHandler
+        Returns:
+            bool: true if the path is a directory
+        """
+        self._check_kwargs(kwargs)
+
+        _, _, dirpath = self._parse_uri(path)
+
+        try:
+            # Enumeration should find at least 1 longer child path
+            blob = next(self._enumerate_blobs(path))
+            return len(blob.name) > len(dirpath)
+        except StopIteration:
+            return False
+        except azure_exceptions.AzureError as e:
+            logger.exception(e)
+            return False
+
+    def _ls(self, path: str, **kwargs: Any) -> List[str]:
+        """
+        List the contents of the directory at the provided URI.
+        Args:
+            path (str): A URI supported by this PathHandler
+        Returns:
+            List[str]: list of contents in given path
+        """
+        self._check_kwargs(kwargs)
+
+        return [blob.name for blob in self._enumerate_blobs(path)]
+
+    def _local_cache_path(self, path: str):
+        """
+        Helper that returns a local cache path for a given uri.
+        Args:
+            path (str): A URI supported by this PathHandler.
+        Returns:
+            local_cache_path (str): a file path which exists on the local file system,
+            in a cache directory.
+        """
+        _, _, blob = self._parse_uri(path)
+        return os.path.join(get_cache_dir(self.cache_dir), self.CACHE_SUBDIR_NAME, blob)
```

Comment:
Nice catch, yeah I noticed this recently. My plan is to set a default in metaseq-internal, but hadn't planned on doing that here or in metaseq. It's hard to say how the package will be used in general so I thought it would be best to let users choose the appropriate location. 

I can log a warning if it falls back to the homedir default. Open to other suggestions as well.

Reply:
Added an issue for follow-up: https://github.com/facebookresearch/iopath/issues/19